baybe/campaign.py

```diff
@@ -222,6 +269,84 @@ def add_measurements(
             numerical_measurements_must_be_within_tolerance,
         )
 
+    def _mark_as_measured(
```

Comment:
I'm fine with moving it's code to `add_measurements`, especially since this mechanism unavoidably needs to be generalized anyway once we continue our path to configurable search space backends or want to switch to polars as main backend. 

Just FYI: what I can imagine that the individual backends could later on provide their own `update_metadata` method where they update an external object that has been created according to their specific needs. But more on that in the future, when we are a bit closer...

Reply:
done